### PR TITLE
math_errhandling_tests: 16-bit int fixes

### PR DESCRIPTION
**Pull request by pjonsson**
Not really sure about this one, I arbitrarily chose 20 as value since the result of the multiplication then fits in a 16-bit int. If the constant 100 is chosen to hit some corner case on 32-bit architectures, my choice of 20 is not the right one.

### CODE REVIEW

File: test/math_errhandling_tests.c [hunk 1]
```diff
@@ -346,13 +346,19 @@ FLOAT_T makemathname(test_jn_snan)(void) { return makemathname(jn)(3, makemathna
 
 #endif
 
+#if INT_MAX == INT16_MAX
```
Comment (keith-packard) [inline test/math_errhandling_tests.c:349]: 20 should work on every target -- it only needs to be larger than `__DBL_MAX_EXP__` by a little bit :-)

Comment (keith-packard) [reply test/math_errhandling_tests.c:349]: > 20 should work on every target -- it only needs to be larger than `__DBL_MAX_EXP__` by a little bit :-)

In other words, let's not have a special case for 16-bit int machines so that we can test the change before we have any of those in CI.

Comment (pjonsson) [reply test/math_errhandling_tests.c]: That's much better, updated the commit with a smaller constant instead.

**Review by keith-packard — APPROVED** [review]
